Mathlib/LinearAlgebra/Multilinear/Basic.lean

```diff
@@ -760,8 +763,150 @@ theorem domDomCongr_eq_iff (σ : ι₁ ≃ ι₂) (f g : MultilinearMap R (fun _
 
 end
 
+
+section Derivative
+
+/-! The goal of this part is, for `f` a multilinear map in finitely many variables indexed by `ι`,
+to develop `f(x + y)` as a finite formal multilinear series in `y`. The term of degree `n` of
+this formal multilinear series will be a sum over finsets `s` of `ι` of cardinality `n` of a
+multilinear map indexed by `s`, evaluated at `fun (_ : s) => y`; we use a linear order on
+`ι` to identify all such `s` to `Fin n`. We also give a more direct definition of the term
+of degree `1` in `f.linearDeriv`, which doesn't require a linear order on `ι`, and prove the
+equivalence of the two definitions.-/
+
+lemma domDomRestrict_aux [DecidableEq ι] (s : Set ι) [(i : ι) → Decidable (i ∈ s)]
+    [DecidableEq s] (x : (i : ↑sᶜ)→ M₁ i) (z : (i : s) → M₁ i) (i : s)
+    (c : M₁ i) : (fun j ↦ if h : j ∈ s then Function.update z i c ⟨j, h⟩ else x ⟨j, h⟩) =
+    Function.update (fun j => if h : j ∈ s then z ⟨j, h⟩ else x ⟨j, h⟩) i c := by
```

Comment:
Possibly:
```suggestion
lemma domDomRestrict_aux [DecidableEq ι] (P : ι → Prop) [DecidablePred p]
    [DecidableEq s] (x : {a // P a} → M₁ i) (z : {a // ¬ P a} → M₁ i) (i : s)
    (c : M₁ i) : (fun j ↦ if h : P j then Function.update x i c ⟨j, h⟩ else z ⟨j, h⟩) =
    Function.update (fun j => if h : j ∈ s then x ⟨j, h⟩ else z ⟨j, h⟩) i c := by
```
(note I also swapped `x` and `z` to put them in the obvious order)

Reply:
Doesn't change much afaik, I made the change.
I also remove the bits about formal multilinear series to apply @alreadydone's suggestion to only define the formal multilinear series explicitly at the origin and then use `changeOrigin` to get it elsewhere. So now all these parts are in the file that proves that continuous multilinear maps are analytic (not yet PR'ed, as I still have to finish some calculations).